src/quic/frame_reader.d

```diff
@@ -0,0 +1,98 @@
+module quic.frame_reader;
+
+import quic.frame;
+import quic.packet;
+import quic.attributes;
+
+struct QuicReader(FrameType)
+{
+    import quic.decode : decodeVarInt;
+
+    ubyte[] networkStream;
+    ulong bufIndex;
+
+    this(ubyte[] networkStream, ulong bufIndex)
+    {
+        this.networkStream = networkStream; 
+        this.bufIndex = bufIndex;
+    }
+
+    auto opDispatch(string name)() {
+            alias FieldType = typeof(mixin(FrameType.stringof, '.', name));
```

Comment:
this code is undefined behavior, instead us `typeof(__traits(getMember, FrameType, name))`

Reply:
(alternatively `typeof(mixin("FrameType." ~ name))` would also work fyi